### ISSUE TITLE
MOFCompiler fails with handle=None

### ISSUE THREAD

**Issue by andy-maier** [pywbem/pywbem#1998]
The `MOFCompiler` class is documented to allow `handle=None` which means that no target repository is used. However, trying to use that fails as follows (using pywbem at master):

```
$ python -c "import pywbem; mc=pywbem.MOFCompiler(handle=None); mc.compile_file('tests/unittest/simple_mock_model.mof', 'root')"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "pywbem/mof_compiler.py", line 2725, in compile_file
    return self.compile_string(mof, ns, filename=filename)
  File "pywbem/mof_compiler.py", line 2669, in compile_string
    self.parser.handle.default_namespace = ns
AttributeError: 'NoneType' object has no attribute 'default_namespace'
```

When fixing that, this further but similar error shows up:

```
$ python -c "import pywbem; mc=pywbem.MOFCompiler(handle=None); mc.compile_file('tests/unittest/simple_mock_model.mof', 'root')"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "pywbem/mof_compiler.py", line 2726, in compile_file
    return self.compile_string(mof, ns, filename=filename)
  File "pywbem/mof_compiler.py", line 2682, in compile_string
    rv = self.parser.parse(mof, lexer=lexer)
  File "/Users/maiera/virtualenvs/pywbem27/lib/python2.7/site-packages/ply/yacc.py", line 333, in parse
    return self.parseopt_notrack(input, lexer, debug, tracking, tokenfunc)
  File "/Users/maiera/virtualenvs/pywbem27/lib/python2.7/site-packages/ply/yacc.py", line 1120, in parseopt_notrack
    p.callable(pslice)
  File "pywbem/mof_compiler.py", line 776, in p_mp_setQualifier
    ns = p.parser.handle.default_namespace
AttributeError: 'NoneType' object has no attribute 'default_namespace'
```

I tested back to pywbem 0.12, and the error always showed up. So the fix should be rolled back into 0.16, but given for how long it did not work, there is no urgency to create a 0.15.1 fix release, IMO.

The support for `handle=None` should be re-established, and a test should be added for that case.

**Comment by KSchopmeyer** [comment]
We agreed that we should make handle=None invalid.  The mininum is some type of local cache for created entities and MOFWBEMConnection is the logical choice as minimum implementation.

**Comment by andy-maier** [comment]
Fixed this differently than agreed, by allowing handle=None and then defaulting it to a MOFWBEMConnection object without a connection. That gives it a local cache for the created entities.

PR TBD fixes this for master, and PR TBD for 0.16.0.
